src/sqlfluff/rules/L009.py

```diff
@@ -95,11 +101,39 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             if file_len != pos:
                 return None
 
-        # We're going to make an edit because we're appending to the end and there's
-        # no segment after it to match on. Otherwise we would never get a match!
-        return LintResult(
-            anchor=context.segment,
-            fixes=[
-                LintFix("edit", context.segment, [context.segment, NewlineSegment()])
-            ],
-        )
+        # Include current segment for complete stack.
+        complete_stack: List[BaseSegment] = list(context.raw_stack)
```

Comment:
This looks like a good fix for the earlier mypy issue. It's a bit confusing that `raw_stack` is a list of `RawSegment` rather than `BaseSegment`. That is, precisely speaking, the correct type, but it frequently requires some fussy type hints or casting because rule code frequently works with other segment types. I _almost_ wonder if we should change `context.raw_stack` to `List[BaseSegment]`. It's still correct, and would require fewer of these casts. (Out of scope for this PR, just thinking...)

Reply:
Agreed, plus at the moment `raw_stack` is returned as a `tuple` whereas a `list` might make more sense for a stack. We can stick with this for now though, given that the type mixing is scoped to this function and mypy would highlight if someone then tried to add a `RawSegment` only bit of code in the next couple lines.